core/app/c/[communitySlug]/pubs/[pubId]/page.tsx

```diff
@@ -166,11 +198,23 @@ export default async function Page(props: {
 		return null;
 	}
 
+	if (!availableViewForms.length) {
+		return null;
+	}
+
 	const pubTypeHasRelatedPubs = pub.pubType.fields.some((field) => field.isRelation);
 	const pubHasRelatedPubs = pub.values.some((value) => !!value.relatedPub);
 
 	const { stage, ...slimPub } = pub;
 	const pubByForm = getPubByForm({ pub, form, withExtraPubValues });
+
+	const editForm =
+		availableUpdateForms.find((form) => form.slug === formSlug) ||
+		availableUpdateForms.find((form) => form.isDefault) ||
+		availableUpdateForms[0];
+
+	const editQuery = editForm.isDefault ? "" : `?form=${editForm.slug}`;
+
```

Comment:
new: make the "update" button on the view page redirect to the same form you're viewing rn, if you have access to it